File: study_ws/src/cpp_pkg/src/news_station.cpp
#include "rclcpp/rclcpp.hpp"
//#include "example_interfaces/msg/string.hpp"
#include "std_msgs/msg/string.hpp"


class news_station : public rclcpp::Node 
{
public:
    news_station() : Node("news")
    {
        publisher_ = this->create_publisher<std_msgs::msg::String>("news", 10);

        this->declare_parameter<std::string>("robot_name_", "R2D2");
        this->declare_parameter<double>("publishFrequency", 1);

        robot_name_ = this->get_parameter("robot_name_").as_string();
        std::double_t delay_ = 1000.0 / this->get_parameter("publishFrequency").get_value<double>();

        timer_ =  this->create_wall_timer(std::chrono::milliseconds(static_cast<int>(delay_)), 
                                         std::bind(&news_station::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "Broadcasting Started...");
    }

private:
    void publishNews(){
        auto msg = std_msgs::msg::String();
        msg.data = "Hello, I'm " + robot_name_ +" :), This is a Test Broadcast..";
        publisher_->publish(msg);
    }
    std::string robot_name_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<news_station>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: study_ws/src/py_pkg/py_pkg/news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class news_station(Node): 
    def __init__(self):
        super().__init__("news_station")
        self.counter = 0
        self.publisher_ = self.create_publisher(String, "news", 10)
        self.get_logger().info("Broadcasting Starts..")
        self.declare_parameter('publishFrequency', 1.0)
        self.delay_ = self.get_parameter('publishFrequency').get_parameter_value().double_value
        self.timer_ = self.create_timer(self.delay_, self.publish_news)
        self.declare_parameter('robot_name', "R2D2")
        self.show_runner_ = self.get_parameter('robot_name').get_parameter_value().string_value

    def publish_news(self):
        msg = String()
        
        #self.counter += 1
        msg.data = "Hi, I'm " + self.show_runner_ + ", This is the Robot News Station.."
        self.publisher_.publish(msg)
        self.get_logger().info(msg.data)

def main(args=None):
    rclpy.init(args=args)
    node = news_station() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
